src/table.js

```diff
@@ -538,15 +543,105 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  const stringValue = typeof value === "string" ? value.trim() : value;
+  switch (type) {
+    case "string":
+      return typeof value === "string"
+        ? stringValue
+        : value || value === 0
+        ? value.toString()
+        : null;
+    case "boolean":
+      return value === true || stringValue === "true"
+        ? true
+        : value === false || stringValue === "false"
+        ? false
+        : null;
+    case "integer":
+      return value === 0
+        ? value
+        : !value || isNaN(parseInt(value))
+        ? NaN
+        : parseInt(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : value === 0 || value === true || value === false
+        ? // eslint-disable-next-line no-undef
+          BigInt(value)
+        : !value || isNaN(value) || !Number.isInteger(+value)
+        ? NaN
+        : // eslint-disable-next-line no-undef
+          BigInt(value);
+    case "number": {
+      return value === 0
+        ? value
+        : !value || isNaN(value)
+        ? NaN
+        : Number(value);
+    }
+    case "date": {
+      if (value instanceof Date) return value;
+      if (typeof value === "string") {
+        let match;
+        if (
+          (match = stringValue.match(
+            /^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?(T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?(Z|[-+]\d{2}:\d{2})?)?$/
+          ))
+        ) {
+          if (fixTz && !!match[4] && !match[7])
+            value = stringValue.replace(/-/g, "/").replace(/T/, " ");
+        }
+      }
+      return new Date(value);
+    }
+    case "array":
+      if (Array.isArray(value)) return value;
+      return [value];
```

Comment:
It feels surprising to me to wrap single values in an array, especially if that value is possibly nullish. I think we’d want to preserve null and undefined here, at least. Or perhaps we can avoid offering `"array"` as a type target for coercion? I don’t see that we need to offer it; it doesn’t make sense for CSV, which is our primary use case for coercion.

Reply:
Also agree that it would make sense to avoid offering `"array"`, `"object"`, `"buffer"`, and `"other"` as target types for coercion.